File: frontend/src/types/submitter.ts
/**
 * Submitter Types
 * 
 * Third-party submission services for grading and authentication.
 */

// ============================================
// SUBMITTER TYPES
// ============================================

export interface Submitter {
  id: string;
  name: string;
  code: string | null;
  website: string | null;
  contact_email: string | null;
  contact_phone: string | null;
  offers_grading: boolean;
  offers_authentication: boolean;
  is_active: boolean;
  is_default: boolean;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface SubmitterCreate {
  name: string;
  code?: string | null;
  website?: string | null;
  contact_email?: string | null;
  contact_phone?: string | null;
  offers_grading?: boolean;
  offers_authentication?: boolean;
  is_active?: boolean;
  is_default?: boolean;
  notes?: string | null;
}

export interface SubmitterUpdate {
  name?: string;
  code?: string | null;
  website?: string | null;
  contact_email?: string | null;
  contact_phone?: string | null;
  offers_grading?: boolean;
  offers_authentication?: boolean;
  is_active?: boolean;
  is_default?: boolean;
  notes?: string | null;
}

export interface SubmitterSummary {
  id: string;
  name: string;
  code: string | null;
  offers_grading: boolean;
  offers_authentication: boolean;
  is_default: boolean;
}

export interface SubmitterStats {
  id: string;
  name: string;
  total_grading_submissions: number;
  total_auth_submissions: number;
  pending_grading: number;
  pending_auth: number;
  cards_graded: number;
  items_authenticated: number;
}

File: backend/app/schemas/submitter.py
"""
Submitter Schemas

Pydantic schemas for third-party submission services.
"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field
from uuid import UUID


# ============================================
# BASE SCHEMAS
# ============================================

class SubmitterBase(BaseModel):
    """Base submitter fields"""
    name: str = Field(..., min_length=1, max_length=100)
    code: Optional[str] = Field(None, max_length=20)
    website: Optional[str] = Field(None, max_length=255)
    contact_email: Optional[str] = Field(None, max_length=255)
    contact_phone: Optional[str] = Field(None, max_length=50)
    offers_grading: bool = True
    offers_authentication: bool = True
    is_active: bool = True
    is_default: bool = False
    notes: Optional[str] = None


# ============================================
# CREATE/UPDATE SCHEMAS
# ============================================

class SubmitterCreate(SubmitterBase):
    """Schema for creating a new submitter"""
    pass


class SubmitterUpdate(BaseModel):
    """Schema for updating a submitter (all fields optional)"""
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    code: Optional[str] = Field(None, max_length=20)
    website: Optional[str] = Field(None, max_length=255)
    contact_email: Optional[str] = Field(None, max_length=255)
    contact_phone: Optional[str] = Field(None, max_length=50)
    offers_grading: Optional[bool] = None
    offers_authentication: Optional[bool] = None
    is_active: Optional[bool] = None
    is_default: Optional[bool] = None
    notes: Optional[str] = None


# ============================================
# RESPONSE SCHEMAS
# ============================================

class SubmitterResponse(SubmitterBase):
    """Full submitter response with all fields"""
    id: UUID
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class SubmitterSummary(BaseModel):
    """Minimal submitter info for dropdowns"""
    id: UUID
    name: str
    code: Optional[str] = None
    offers_grading: bool
    offers_authentication: bool
    is_default: bool
    
    class Config:
        from_attributes = True


# ============================================
# STATS SCHEMA
# ============================================

class SubmitterStats(BaseModel):
    """Statistics for a submitter"""
    id: UUID
    name: str
    total_grading_submissions: int = 0
    total_auth_submissions: int = 0
    pending_grading: int = 0
    pending_auth: int = 0
    cards_graded: int = 0
    items_authenticated: int = 0